src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantAdvertService.java

```diff
@@ -207,19 +205,47 @@ private void addStaticTimeToDateQuestion(GrantAdvertPageResponseValidationDto pa
             pagePatchDto.getPage().getQuestions().forEach(question -> {
                 String[] multiResponse = question.getMultiResponse();
                 if (question.getId().equals(OPENING_DATE_ID)) {
-                    String[] opening = new String[] { multiResponse[0], multiResponse[1], multiResponse[2], "00",
-                            "01" };
+                    String[] openingTime = multiResponse[3].split(":");
+                    String[] opening = new String[] { multiResponse[0], multiResponse[1], multiResponse[2],
+                            openingTime[0], openingTime[1] };
                     pagePatchDto.getPage().getQuestions().get(0).setMultiResponse(opening);
                 }
                 else if (question.getId().equals(CLOSING_DATE_ID)) {
-                    String[] closing = new String[] { multiResponse[0], multiResponse[1], multiResponse[2], "23",
-                            "59" };
+                    String[] closingTime = multiResponse[3].split(":");
+                    String[] closing;
+
+                    if (multiResponse[3].equals("23:59")) {
+                        // increment date by 1 day and set time to 00:00
+                        LocalDateTime dateTime = convertToDateTime(multiResponse, closingTime);
+                        LocalDateTime incrementedDateTime = dateTime.plusDays(1);
+                        int incrementedDay = incrementedDateTime.getDayOfMonth();
+                        int incrementedMonth = incrementedDateTime.getMonthValue();
+                        int incrementedYear = incrementedDateTime.getYear();
+
+                        closing = new String[] { String.format("%02d", incrementedDay),
+                                String.valueOf(incrementedMonth), String.valueOf(incrementedYear), "00", "00" };
+                    }
+                    else {
+                        closing = new String[] { multiResponse[0], multiResponse[1], multiResponse[2], closingTime[0],
+                                closingTime[1] };
+                    }
```

Comment:
```suggestion
                    String[] closingDateTime = new String[] { multiResponse[0], multiResponse[1], multiResponse[2],
                            closingTime[0], closingTime[1] };

                    if (multiResponse[3].equals("23:59")) {
                        // increment date by 1 day and set time to 00:00
                        LocalDateTime dateTime = convertToDateTime(multiResponse, closingTime);
                        LocalDateTime incrementedDateTime = dateTime.plusDays(1);
                        int incrementedDay = incrementedDateTime.getDayOfMonth();
                        int incrementedMonth = incrementedDateTime.getMonthValue();
                        int incrementedYear = incrementedDateTime.getYear();

                        closingDateTime = new String[] { String.format("%02d", incrementedDay),
                                String.valueOf(incrementedMonth), String.valueOf(incrementedYear), "00", "00" };
                    }
                    }
                    else {
                        closing = new String[] { multiResponse[0], multiResponse[1], multiResponse[2], closingTime[0],
                                closingTime[1] };
                    }
```

Reply:
Might also be worth extracting the body if this if statement to a function named something like `adjustToMidnightNextDay()`